### PR TITLE
[ECO-378] Remove unnecessary mutable borrows in view funcs

### PR DESCRIPTION
**Pull request by alnoki**
@chen-robert

This PR removes unnecessary mutable borrows in view functions.

Note that https://github.com/aptos-labs/aptos-core/pull/9369 changed an event unit testing function signature, and so tests are updated here to enable a test run.

All tests pass.

### CODE REVIEW

File: src/move/econia/sources/registry.move [hunk 1]
```diff
@@ -2481,7 +2481,7 @@ module econia::registry {
         assert!(get_market_id_base_coin<BC, QC>(
                     lot_size_2, tick_size_2, min_size_2) == option::none(), 0);
         // Assert events.
-        let market_registration_events = event::emitted_events(
+        let market_registration_events = event::emitted_events_by_handle(
```
Comment (elliottdehn) [inline src/move/econia/sources/registry.move:2484]: These changes seem unrelated?

File: src/move/econia/sources/incentives.move [hunk 2]
```diff
@@ -566,17 +566,17 @@ module econia::incentives {
         IncentiveParameters,
         IntegratorFeeStores
     {
-        // Borrow mutable reference to integrator fee stores map for
-        // given quote coin type.
-        let integrator_fee_stores_map_ref_mut =
-            &mut borrow_global_mut<IntegratorFeeStores<QuoteCoinType>>(
+        // Immutably borrow integrator fee stores map for given quote
+        // coin type.
+        let integrator_fee_stores_map_ref =
```
Comment (elliottdehn) [inline src/move/econia/sources/incentives.move:571]: Maybe they should add this as a warning in the move compiler: unused mutable attribute

**Review by elliottdehn — APPROVED** [review]

File: src/move/econia/sources/incentives.move (hunk 2 above)
Comment (alnoki) [reply src/move/econia/sources/incentives.move:571]: Agreed

File: src/move/econia/sources/registry.move (hunk 1 above)
Comment (alnoki) [reply src/move/econia/sources/registry.move:2484]: Yes, but the tests in the PR can't run without fixing this due to upstream changes forced by Aptos Labs
